x-pack/plugins/enterprise_search/public/applications/app_search/components/search/search_logic.test.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import '../../__mocks__/engine_logic.mock';
+
+import { LogicMounter, mockHttpValues, mockFlashMessageHelpers } from '../../../__mocks__';
+
+import { nextTick } from '@kbn/test/jest';
+
+import { SearchLogic } from './search_logic';
+
+describe('SearchLogic', () => {
+  const { mount } = new LogicMounter(SearchLogic);
+  const { http } = mockHttpValues;
+  const { flashAPIErrors } = mockFlashMessageHelpers;
+
+  const MOCK_SEARCH_RESPONSE = {
+    results: [
+      { id: { raw: 'document-1' }, _meta: { id: 'document-1', engine: 'some-engine' } },
+      { id: { raw: 'document-2' }, _meta: { id: 'document-2', engine: 'some-engine' } },
+      { id: { raw: 'document-3' }, _meta: { id: 'document-3', engine: 'some-engine' } },
+    ],
+  };
+
+  const DEFAULT_VALUES = {
+    searchDataLoading: false,
```

Comment:
I named this `searchDataLoading` rather than `dataLoading` so that it can exist alongside other logic files that may use `dataLoading` already.

Reply:
Yeah. It's just a slight pain if you have something like:

```
const { dataLoading } = useValues(SearchLogic)
const { dataLoading } = useValues(EngineLogic)
```